kernel/kernel-api/src/main/java/io/delta/kernel/internal/ScanImpl.java

```diff
@@ -151,19 +153,36 @@ private CloseableIterator<FilteredColumnarBatch> applyPartitionPruning(
             partitionPredicate.get(),
             partitionColNameToTypeMap);
 
-        PredicateEvaluator predicateEvaluator =
```

Comment:
why expand the `Iterator.map`? We can make the `predicateEvaluator` create an optional in the current code itself.

Reply:
Need the schema of the batch to create the predicate evaluator (since it may or may not have the `add.stats` column)